org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AspectjFilter.java

```diff
@@ -0,0 +1,206 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Lars Grefer - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import org.objectweb.asm.Attribute;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.JumpInsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+
+import java.util.regex.Pattern;
+
+/**
+ * Filters synthetic methods created by the AspectJ-Compiler (ajc)
+ * <p>
+ * Every class containing an advice, will get the synthetic methods {@code aspectOf(..)},
+ * {@code hasAspect(..)}. If the advice is a singleton, the class will also get the
+ * {@code ajc$postClinit()} method which will be called from the static initializer
+ * ({@code <clinit>()}).
+ * This filter ignores these 3 method and the call.
+ * <p>
+ * Every class containing advised methods, where the advice takes a JoinPoint as parameter,
+ * will contain the {@code ajc$preClinit()} method which will be called from the static
+ * initializer ({@code <clinit>()}).
+ * This filter ignores this method, and its call.
+ * <p>
+ * If the static initializer ({@code <clinit>()}) only contains the calls to
+ * {@code ajc$preClinit()} and/or {@code ajc$postClinit()}, its completely ignored.
+ * <p>
+ * For every JoinPoint which is advised by an around advice, an {@code $AjcClosure}-class
+ * is generated (unless the around advice is inlined).
+ * This filter ignored all methods ins such classes.
+ */
+public class AspectjFilter implements IFilter {
+
+    private static final String AJ_SYNTHETIC_ATTRIBUTE = "org.aspectj.weaver.AjSynthetic";
+    private static final Pattern AJC_CLOSURE_PATTERN = Pattern.compile(".*\\$AjcClosure\\d+");
+
+    public void filter(MethodNode methodNode, IFilterContext context, IFilterOutput output) {
+
+        if (isAroundClosureClass(context)) {
+            output.ignore(methodNode.instructions.getFirst(), methodNode.instructions.getLast());
+            return;
+        }
+
+        if (isWellKnownSyntheticMethod(methodNode)) {
+            output.ignore(methodNode.instructions.getFirst(), methodNode.instructions.getLast());
+            return;
+        }
+
+        if (methodNode.name.equals("<clinit>")) {
+            checkStaticInitializer(methodNode, output);
+        }
+    }
+
+    private boolean isWellKnownSyntheticMethod(MethodNode methodNode) {
+        if (methodNode.name.equals("aspectOf") && isAjSynthetic(methodNode)) {
+            return true;
+        }
+        if (methodNode.name.equals("hasAspect") && isAjSynthetic(methodNode)) {
+            return true;
+        }
+        if (methodNode.name.equals("ajc$postClinit") && isAjSynthetic(methodNode)) {
```

Comment:
What's the point of this?

According to [your example](https://github.com/larsgrefer/aspectj-jacoco-example/tree/afcea2a2a0d7ab78f5fef26d7ec26c7f7f0a129e) (https://github.com/jacoco/jacoco/pull/710#issuecomment-404971814) `ajc$postClinit` has `ACC_SYNTHETIC` flag

```
$ javap -v -p build/classes/java/main/de/larsgrefer/demo/Combined.class | grep -A 2 "void ajc\$postClinit"
  private static void ajc$postClinit();
    descriptor: ()V
    flags: ACC_PRIVATE, ACC_STATIC, ACC_SYNTHETIC
```

and therefore already filtered by `SyntheticFilter`.

Reply:
You're right. This method is already filtered. I've simplified the AspectJ filter accordingly.